### PR TITLE
Eagle 198  integrate new Siddhi version so that thread should not die when event evaluation fails

### PR DESCRIPTION
**Pull request by yonzhang**
when exception happens within Siddhi processing thread, it can continue to work

### CODE REVIEW

File: eagle-core/eagle-policy/eagle-policy-base/src/main/java/org/apache/eagle/policy/siddhi/SiddhiPolicyEvaluator.java [hunk 1]
```diff
@@ -130,6 +133,7 @@ private SiddhiRuntime createSiddhiRuntime(SiddhiPolicyDefinition policyDef) {
 
         try {
             executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(executionPlan);
+            executionPlanRuntime.handleExceptionWith(new SiddhiPolicyExceptionHandler());
```
Comment (haoch) [inline eagle-core/eagle-policy/eagle-policy-base/src/main/java/org/apache/eagle/policy/siddhi/SiddhiPolicyEvaluator.java]: My patch is to extend capability of customizing exception handler for siddhi while it uses FatalExceptionHandler by default, so here we should override our own exception handler `SiddhiPolicyExceptionHandler` and explicitly specify by `handleExceptionWith`.

**Comment by haoch** [comment]
Another problem is Ci failed due to 

~~~
Tests in error: 
  testExternalTimeSlideWindowWithGroupby(org.apache.eagle.alert.state.TestSiddhiStateSnapshotAndRestore)

Tests run: 37, Failures: 0, Errors: 1, Skipped: 1
~~~

**Comment by yonzhang** [comment]
now after 3.0.5, the default exception handler is not FatalExceptionHandler, do we still need SiddhiPolicyExceptionHandler?

**Comment by haoch** [comment]
If so, then we don't need to customize SiddhiPolicyExceptionHandler then.

**Pull request merged**